zappa/zappa.py

```diff
@@ -923,13 +961,13 @@ def undeploy_api_gateway(self, project_name, api_key_required=False, domain_name
                     restApiId=api['id']
             )
 
-            if api_key_required:
-                print("Removing API Key..")
-                api_key_id = [key for key in self.apigateway_client.get_api_keys()['items']
-                              if api['id'] in key['name']][0]['id']
-                api_key_response = self.apigateway_client.delete_api_key(
-                        apiKey="{}".format(api_key_id)
-                )
+            if remove_api_key:
+                try:
+                    api_key_id = [key for key in self.apigateway_client.get_api_keys()['items']
```

Comment:
This line is really, really ugly - can we refactor this into something more readable?